File: FrontEnd/src/app/features/sales/invoicesCommon/view-models/pdf/invoice-pdf-vm.ts
import { InvoicePdfAadeVM } from './invoice-pdf-aade-vm'
import { InvoicePdfBankAccountVM } from './invoice-pdf-bankAccount-vm'
import { InvoicePdfHeaderVM } from './invoice-pdf-header-vm'
import { InvoicePdfPartyTypeVM } from './invoice-pdf-partyType-vm'
import { InvoicePdfPortVM } from './invoice-pdf-port-vm'
import { InvoicePdfShipVM } from './invoice-pdf-ship-vm'
import { InvoicePdfSummaryVM } from './invoice-pdf-summary-vm'

export interface InvoicePdfVM {

    header: InvoicePdfHeaderVM
    issuer: InvoicePdfPartyTypeVM
    customer: InvoicePdfPartyTypeVM
    aade: InvoicePdfAadeVM
    ports: InvoicePdfPortVM[]
    ship: InvoicePdfShipVM
    paymentMethod: string
    bankAccounts: InvoicePdfBankAccountVM[]
    summary: InvoicePdfSummaryVM
    previousBalance: number
    newBalance: number

}

File: API/Features/Sales/Invoices/ViewModels/Pdf/InvoicePdfVM.cs
using System;
using System.Collections.Generic;
using API.Infrastructure.Classes;

namespace API.Features.Sales.Invoices {

    public class InvoicePdfVM {

        public Guid InvoiceId { get; set; }
        public InvoicePdfHeaderVM Header { get; set; }
        public string Remarks { get; set; }
        public InvoicePdfShipVM Ship { get; set; }
        public InvoicePdfAadeVM Aade { get; set; }
        public List<InvoicePdfPortVM> Ports { get; set; }
        public InvoicePdfPartyVM Customer { get; set; }
        public string Destination { get; set; }
        public InvoicePdfDocumentTypeVM DocumentType { get; set; }
        public string PaymentMethod { get; set; }
        public InvoicePdfPartyVM Issuer { get; set; }
        public InvoicePdfSummaryVM Summary { get; set; }
        public decimal PreviousBalance { get; set; }
        public decimal NewBalance { get; set; }
        public SimpleEntity[] BankAccounts { get; set; }

    }

}